### PR TITLE
Bug fix/resilience hangup

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Possible bugfix for https://github.com/arangodb/release-3.6/issues/5.

**TODO** Clean up this description.

*(Can you describe what functional change your PR is trying to effect?)*

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [ ] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [ ] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [ ] The behavior change can be verified via automatic tests

#### Related Information

*(Please reference tickets / specification etc )*

- [ ] There is a GitHub Issue reported by a Community User: 
- [ ] There is an internal planning ticket: 
- [ ] There is a *JIRA Ticket number* (In case a customer was affected / involved): 
- [ ] There is a design document: 

### Testing & Verification

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This PR adds tests that were used to verify all changes:

- [ ] Added **Regression Tests** (Only for bug-fixes) 
- [ ] Added new C++ **Unit Tests** (Either GoogleTest or Catch-Test)
   - Did you add tests for a new *RestHandler* subclass ?
   - Did you add new mocks of underlying code layers to be able to verify your functionality ? 
   - ...
- [ ] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
- [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

Additionally:

- [ ] There are tests in an external testing repository (i.e. node-resilience tests, chaos tests)
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

*(Include link to Jenkins run etc)*

> Think about whether the new code you added is modular enough to be
> easily testable by unit tests written with GTest / Catch. It is not good if your feature is so interconnected
> that it prevents other people from writing their own unit gests. It should be possible
> to use your code in future without extensively mocking your classes.
> A bad example that required some extensive effort would be the storage engine API.

### Documentation

> All new Features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the changelog so that
> developers and users have a concise overview. 

- [ ] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *http API* 
- [ ] Added *Swagger examples* for the http API  

### CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7729/

**Comment by goedderz** [comment]
There is one test failing with mmfiles/single. Need to investigate.

File: arangod/Transaction/Helpers.cpp [hunk 1]
```diff
@@ -402,9 +404,17 @@ std::string transaction::helpers::makeIdFromCustom(CollectionNameResolver const*
   TRI_ASSERT(id.isCustom() && id.head() == 0xf3);
   TRI_ASSERT(key.isString());
 
-  uint64_t cid = encoding::readNumber<uint64_t>(id.begin() + 1, sizeof(uint64_t));
+  auto cid = encoding::readNumber<uint64_t>(id.begin() + 1, sizeof(uint64_t));
 
   std::string resolved = resolver->getCollectionNameCluster(cid);
+  if (resolved == "_unknown") {
```
Comment (graetzer) [inline arangod/Transaction/Helpers.cpp]: This is so bad

Comment (graetzer) [reply arangod/Transaction/Helpers.cpp]: Are you really gonna compare against that magic string?  This is a code smell

Comment (goedderz) [reply arangod/Transaction/Helpers.cpp]: No, you're right of course. This was originally just an experiment whether it helped at all, and I've failed to clean it up properly. And I'd still classify this PR as experimental.

However, the whole thing is just a hack, and we should never need to load the plan during a velocypack dump; but that is a larger fix and will need to be done separately.

**Pull request closed without merging**